src/docs/release/breaking-changes/layout-builder-optimization.md

```diff
@@ -0,0 +1,170 @@
+case. If the constraints are the same and the widget configuration did not
+change, the builder function is not called.
+
+Your app can break if it relies on the relayout to cause the rebuilding of the
+`LayoutBuilder` rather than on an explicit call to `setState`. This usually
+happens by accident. You meant to add `setState`, but you forgot because the app
+continued functioning as you wanted, and therefore nothing reminded you to add
+it.
+
+## Migration guide
+
+Look for usages of `LayoutBuilder` and `SliverLayoutBuilder` and make sure to
+call `setState` any time the widget state changes.
+
+**Example**: in the example below the contents of the builder function depend
+on the value of the `_counter` field. Therefore, whenever the value is updated,
+you should call `setState` to tell the framework to rebuild the widget. However,
+this example may have previously worked even without calling `setState`, if the
+`_ResizingBox` triggers a relayout of `LayoutBuilder` (see the implementation
+of `_ResizingBox` in the [design doc][1]).
+
+Code before migration (note the missing `setState` inside the `onPressed`
+callback):
+
+```dart
+class Counter extends StatefulWidget {
+  Counter({Key key}) : super(key: key);
+
+  @override
+  _CounterState createState() => _CounterState();
+}
+
+class _CounterState extends State<Counter> {
+  int _counter = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(child: Container(
+      child: LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) {
+          return _ResizingBox(
+            FlatButton(onPressed: () {
+              _counter++;
+            }, child: Text('Increment Counter')),
+            Text(_counter.toString()),
+          );
+        },
+      ),
+    ));
+  }
+}
+```
+
+Code after migration (`setState` added to `onPressed`):
+
+```dart
+class Counter extends StatefulWidget {
+  Counter({Key key}) : super(key: key);
+
+  @override
+  _CounterState createState() => _CounterState();
+}
+
+class _CounterState extends State<Counter> {
+  int _counter = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(child: Container(
+      child: LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) {
+          return _ResizingBox(
+            FlatButton(onPressed: () {
+              setState(() {
+                _counter++;
+              });
+            }, child: Text('Increment Counter')),
+            Text(_counter.toString()),
+          );
+        },
+      ),
+    ));
+  }
+}
+```
+
+Watch for usages of `Animation` and `LayoutBuilder` in the same widget.
+Animations have internal mutable state that changes on every frame. If the
+logic of your builder function depends on the value of the animation, it may
+require a `setState` to update in tandem with the animation. To do that, add an
+[animation listener][7] that calls `setState`, like so:
+
```

Comment:
This code is also not a complete example and is breaking Travis, so add the <!-- skip --> line.

Reply:
Done (skipped)